### PR TITLE
[MeshMoving] Fix Race Condition In MoveMesh

### PR DESCRIPTION
**Pull request by matekelemen**
## Main change
Move the `ParametricAffineTransform` in `block_for_each` from a lambda capture to TLS, because it's not thread-safe. The related discussion is in #11804.

## Other changes
- cosmetic changes
- `LinearTransform` was a misnomer (because it also involved a translation) so I renamed it to `AffineTransform`

### CODE REVIEW

File: applications/MeshMovingApplication/custom_utilities/move_mesh_utilities.cpp [hunk 1]
```diff
@@ -115,17 +116,18 @@ void MoveModelPart(
...
         rModelPart.Nodes(),
-        [&rTransform, time](Node& rNode){
+        rTransform,
+        [time](Node& rNode, ParametricAffineTransform& rTLSTransform){
             const array_1d<double,3>& initial_position = rNode.GetInitialPosition();
-            noalias(rNode.GetSolutionStepValue(MESH_DISPLACEMENT)) = rTransform.Apply(
+            noalias(rNode.GetSolutionStepValue(MESH_DISPLACEMENT)) = rTLSTransform.Apply(
                 initial_position,
                 time,
                 rNode.X0(),
```
Comment (matekelemen) [inline applications/MeshMovingApplication/custom_utilities/move_mesh_utilities.cpp:133]: this is the main change, the rest is style/docs/renaming

File: applications/MeshMovingApplication/custom_utilities/affine_transform.h [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:        BSD License
+//	                Kratos default license: kratos/license.txt
+//
+//  Main Authors:   Máté Kelemen
+//
+
+#ifndef KRATOS_MESH_MOVING_LINEAR_TRANSFORMATION_INCLUDED
```
Comment (loumalouomega) [inline applications/MeshMovingApplication/custom_utilities/affine_transform.h]: pragma once

**Review by RiccardoRossi — DISMISSED** [review]
Overall i am ok with the change, however i see you rename methods so this makes it an API changer.

are u sure those methods are not used by other people?
If that is so (no impact on other people) than just merge...otherwise let's discuss the way forward to minimize impact onto other people

**Comment by matekelemen** [comment]
> Overall i am ok with the change, however i see you rename methods so this makes it an API changer.
> 
> are u sure those methods are not used by other people? If that is so (no impact on other people) than just merge...otherwise let's discuss the way forward to minimize impact onto other people

I'm not sure how many people are using it. I originally wrote these ~3 years ago for @mpentek. Maybe he has a better idea of other potential users.

**Review by philbucher — DISMISSED** [review]
I dont think we need the backward compatibility

**Review by loumalouomega — APPROVED** [review]

**Review by philbucher — APPROVED** [review]

**Comment by roigcarlo** [comment]
Can we merge?

**Comment by RiccardoRossi** [comment]
go ahead

**Pull request merged**